arangod/Pregel/Worker.cpp

```diff
@@ -168,9 +168,16 @@ void Worker<V, E, M>::setupWorker() {
     // of time. Therefore this is performed asynchronous
     TRI_ASSERT(SchedulerFeature::SCHEDULER != nullptr);
     Scheduler* scheduler = SchedulerFeature::SCHEDULER;
-    auto self = shared_from_this();
-    bool queued = scheduler->queue(RequestLane::INTERNAL_LOW, [self, this, cb] {
-      _graphStore->loadShards(&_config, cb);
+    bool queued = scheduler->queue(RequestLane::INTERNAL_LOW, [self = shared_from_this(), this, cb] {
```

Comment:
This `shared_from_this` confused me, because the lambda captured `cb` which itself also holds a shared_reference to `this`. So you could drop one of them? In particular in the case of `lazyLoading()` `cb` is executed immediately. And afaik having `self` and `this` prevents small function optimization and thus allocated memory?

Reply:
Fixed